README.md

```diff
@@ -30,18 +30,108 @@ Input JSON file with road OpenStreetMap tags:
 }
 ```
 
-Output lane specifications from left to write:
+Output: A list of lanes from left to write according to the [specification](./data/spec-lanes.json).
+JSON formatted example:
 
 ```json
 [
-    {"type": "sidewalk", "direction": "backward"},
-    {"type": "cycleway", "direction": "forward"},
+    {"type": "travel_lane", "direction": "backward"},
+    {"type": "travel_lane", "direction": "backward"},
     {"type": "travel_lane", "direction": "forward"},
     {"type": "travel_lane", "direction": "forward"},
     {"type": "sidewalk", "direction": "forward"}
 ]
 ```
 
+Lane Definition
+---------------
+
+It is important to determine a strict definition for a lane.
+An OpenStreetMap way, usually a `highway=*`, can in some cases be divided into lanes,
+where each lane extends from the start to the end of the way.
```

Comment:
"where each lane extends from the start to the end of the way."

I've been puzzling over this one for a long time in the context of street parking. A simple issue is that a parking "lane" is regularly interrupted for driveways. The OSM wiki (I could dig up a reference) does _not_ recommend splitting the way constantly to deal with this, and in practice nobody does, which I think is good. It may be something we want to clarify in the spec -- in this case, a parking lane's use is split between storing vehicles and keeping driveways clear.

It gets more complicated with "pocket parking" and "bulb-outs". One image I found is https://www.susdrain.org/images/derbyshirestreet_fig1.jpg, but I could go take better pictures. Many streets change width frequently on one block.

So far, I think we've been choosing to sacrifice this level of detail, because mapping it is hard in practice and it greatly complicates any renderers/tools built on top.

Just wanted to raise this issue generally, see if you have thoughts on a better representation

Reply:
hmm, as with everything in OSM, we should just support some incremental addition of detail:

1. Some way
2. parking=yes
3. parking=parallel_excluding_driveways (or something) and any other common arrangements
4. micromapping the parking spots as areas, no longer attached to the way? This will add complexity when we need to stitch those back together with the roads, so I suspect this will only come after we have a good way of doing road areas...